app/src/main/java/com/duckduckgo/app/global/view/DaxDialog.kt

```diff
@@ -0,0 +1,147 @@
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.global.view
+
+import android.app.Dialog
+import android.content.DialogInterface
+import android.graphics.Color
+import android.graphics.Rect
+import android.graphics.drawable.ColorDrawable
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.DialogFragment
+import com.duckduckgo.app.browser.R
+import android.view.Gravity
+import androidx.constraintlayout.widget.ConstraintLayout
+import kotlinx.android.synthetic.main.content_dax_dialog.*
+import android.view.animation.Animation
+import android.view.animation.OvershootInterpolator
+import android.view.animation.ScaleAnimation
+import android.widget.ImageView
+import android.widget.RelativeLayout
+
+class DaxDialog(
+    private val daxText: String,
+    private val buttonText: String,
+    private val dismissible: Boolean = true,
+    private val typingDelayInMs: Long = 20
+) : DialogFragment() {
+
+    private var onAnimationFinished: () -> Unit = {}
+    private var primaryCTAClickListener: () -> Unit = { dismiss() }
+    private var hideClickListener: () -> Unit = { dismiss() }
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? =
+        inflater.inflate(R.layout.content_dax_dialog, container, false)
+
+    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+        val dialog = super.onCreateDialog(savedInstanceState)
+        val window = dialog.window
+        val attributes = window?.attributes
+
+        attributes?.gravity = Gravity.BOTTOM
+        window?.attributes = attributes
+        window?.setBackgroundDrawable(ColorDrawable(Color.TRANSPARENT))
+
+        return dialog
+    }
+
+    override fun onStart() {
+        super.onStart()
+        setDialog()
+        setListeners()
+        dialogText.startTypingAnimation(daxText, true, onAnimationFinished)
+    }
+
+    override fun onCancel(dialog: DialogInterface) {
+        dialogText.cancelAnimation()
+        super.onCancel(dialog)
+    }
+
+    fun onAnimationFinishedListener(onAnimationFinished: () -> Unit) {
+        this.onAnimationFinished = onAnimationFinished
+    }
+
+    fun setPrimaryCTAClickListener(clickListener: () -> Unit) {
+        primaryCTAClickListener = clickListener
+    }
+
+    fun setHideClickListener(clickListener: () -> Unit) {
+        hideClickListener = clickListener
+    }
+
+    fun startHighlightViewAnimation(targetView: View, duration: Long = 400, size: Float = 6f) {
+        val highlightImageView = addHighlightView(targetView, size)
+        val fadeInAnimation = ScaleAnimation(0f, 1f, 0f, 1f, Animation.RELATIVE_TO_SELF, 0.5f, Animation.RELATIVE_TO_SELF, 0.5f)
+        fadeInAnimation.duration = duration
+        fadeInAnimation.fillAfter = true
+        fadeInAnimation.interpolator = OvershootInterpolator(3f)
```

Comment:
nit: `3f` is a magic number here - consider replacing with a `const` which explains it

Reply:
Done!